crates/cli/src/subcommands/publish.rs

```diff
@@ -111,6 +112,8 @@ pub async fn exec(mut config: Config, args: &ArgMatches) -> Result<(), anyhow::E
     let build_debug = args.get_flag("debug");
     let wasm_file = args.get_one::<PathBuf>("wasm_file");
     let database_host = config.get_host_url(server)?;
+    // Validate the identity first to provide an early error message
+    util::select_identity_config(&mut config, identity, server).await?;
```

Comment:
could we just move the later code to here, rather than effectively duplicating this check? like could we put this up here:
```
    let (auth_header, identity) = get_auth_header(&mut config, anon_identity, identity, server)
        .await?
        .unzip();
```